### DISCUSSION TITLE
How do different layouts ??

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1234] [General]
I am using layout system in my application which is written in PyQt
Can you explain how we can implement following in Cogent-Core
HBoxLayout 
VBoxLayout 
GridLayout 
FormLayout

**Comment by kkoreilly** [comment]
If you do `core.NewFrame`, then the default should be similar to HBoxLayout. To make it vertical (VBoxLayout), you can do `s.Direction = styles.Column` in a `Styler`. To make it a grid, you can do `s.Display = styles.Grid`. To make a form for editing a struct, you can do `core.NewForm`, or you can just do `s.Display = styles.Grid` and `s.Columns = 2` if you don't want to use a struct.

**Reply by heejit** [reply]
okay thanks
